### PR TITLE
add ssh flags

### LINKED ISSUE #55: Add `ssh-flags` argument to allow customization of ssh execution
Currently `ssh-cmd` argument does not work due to `which` command usage. Probably flags should include key by default.

### PR DESCRIPTION
**Pull request by idruzhitskiy**
Closes #55

### CODE REVIEW

File: src/BobToolsCli/CommonWithSshArguments.cs [hunk 1]
```diff
@@ -8,6 +10,9 @@ public class CommonWithSshArguments : CommonArguments
         [Option("ssh-cmd", HelpText = "Ssh cmd.", Default = "ssh")]
         public string SshCmd { get; set; }
 
+        [Option("ssh-flags", HelpText = "Additional flags to pass to ssh. Without spaces, without -, delimeted by comma", Separator = ',')]
```
Comment (ikopylov) [inline src/BobToolsCli/CommonWithSshArguments.cs:13]: Add an example to `HelpText`

File: src/RemoteFileCopy/Ssh/SshConfiguration.cs [hunk 2]
```diff
@@ -1,22 +1,27 @@
...
     {
         public SshConfiguration(
             string cmd,
+            IEnumerable<string> flags,
             int port,
             string username,
             string pathToKey)
         {
             Cmd = cmd;
+            Flags = flags;
```
Comment (ikopylov) [inline src/RemoteFileCopy/Ssh/SshConfiguration.cs:15]: I suggest to materialize flags into collection here (array or List)

File: src/RemoteFileCopy/Ssh/SshConfiguration.cs [hunk 3]
```diff
@@ -1,22 +1,28 @@
...
         {
             Cmd = cmd;
+            Flags = flags.ToArray();
             Port = port;
             Username = username;
             PathToKey = pathToKey;
         }
 
         public string Cmd { get; }
+        public string[] Flags { get; }
```
Comment (ikopylov) [inline src/RemoteFileCopy/Ssh/SshConfiguration.cs:23]: Maybe it is better to return `IReadOnlyList<string>`

Comment (idruzhitskiy) [reply src/RemoteFileCopy/Ssh/SshConfiguration.cs]: Not sure about losing type info here, but ok

File: src/RemoteFileCopy/Ssh/SshConfiguration.cs [hunk 4]
```diff
@@ -1,22 +1,28 @@
...
         {
             Cmd = cmd;
+            Flags = flags.ToArray();
             Port = port;
             Username = username;
             PathToKey = pathToKey;
         }
 
         public string Cmd { get; }
+        public IReadOnlyCollection<string> Flags { get; }
```
Comment (ikopylov) [inline src/RemoteFileCopy/Ssh/SshConfiguration.cs:23]: `IReadOnlyList<string>` is better, because it allow access by index

**Review by ikopylov — APPROVED** [review]
LGTM

**Pull request merged**